fix(rest): Added code to get obligation releaseView data in project.

nikkuma7 opened the pull request:
<!-- Before filling this issue, please read the wiki (https://eclipse.org/sw360)
and search if the bug do not already exists in the issues (https://github.com//eclipse-sw360/sw360/issues). -->

### Description

fix(rest): Added code to get obligation releaseView data in project.

http://localhost:8080/resource/api/projects/{ id}/licenseObligations?view=true

Closes : #2485

Please describe your issue in few words here.

#### How to reproduce

Describe the bug and list the steps you used when the issue occurred.

#### Screenshots ( if applicable )

Add screenshots to help explain your problem.

### Versions

* Docker version OR
* Last commit id on main

#### SW360 logs

* With docker through `docker logs sw360`
* From bare metal / vm install system under /var/log/sw360/sw360.log and /var/log/tomcat/error.log

GMishx commented:
@nikkuma7 please rebase with latest main

GMishx commented on rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:
```diff
@@ -2284,13 +2284,14 @@ private Map<String, Object> createPaginationMetadata(Pageable pageable, Map<Stri
-            @Parameter(description = "Project ID.") @PathVariable("id") String id)
+	public ResponseEntity<Object> getLicenseObligations(Pageable pageable,
+            @Parameter(description = "Project ID.") @PathVariable("id") String id,
+            @RequestParam(value = "view", defaultValue = "false") boolean releaseView)
```
Please add documentation with `@Parameter` annotation for `releaseView`.

GMishx commented on rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java:
```diff
@@ -905,5 +907,67 @@ public Integer loadPreferredClearingDateLimit() {
+
+        for (Release release : releases) {
+            List<Attachment> approvedCliAttachments = SW360Utils.getApprovedClxAttachmentForRelease(release);
+            if (approvedCliAttachments.isEmpty()) {
+                approvedCliAttachments = SW360Utils.getClxAttachmentForRelease(release);
+            }
```
Maybe put a notice in log that no approved CLX attached found. Going with attached CLX.

nikkuma7 replied:
comment addressed.

GMishx commented on rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java:
```diff
@@ -953,5 +955,72 @@ public Integer loadPreferredClearingDateLimit() {
+            }
+            final String releaseId = release.getId();
+
+            if (approvedCliAttachments.size() == 1) {
```
What will happen if there are more than 1 approved CLI attached? What is expected here?

nikkuma7 replied:
done.

GMishx reviewed (dismissed):
Changes looks good.

GMishx commented:
@deo002 please test this PR.

deo002 commented:
status value is null for obligations in response

nikkuma7 commented:
> status value is null for obligations in response

![image](https://github.com/user-attachments/assets/3a2db8c8-308b-483c-ab72-edf7f18898d6)

Addressed comment.

GMishx commented:
@deo002 please test this PR again.

@nikkuma7 please rebase.

deo002 commented:
Tested, working as expected.
![image](https://github.com/user-attachments/assets/e8da784a-036d-4d49-b86f-0cff69c443da)

GMishx reviewed (approved):
Tests approved by @deo002
